### PR TITLE
Login and signup error handling

### PR DESCRIPTION
**Pull request by komalahuwalia**
Before your pull request is reviewed and merged, please ensure that:

* [ ] there are no linting errors
* [ ] your code is in a uniquely-named feature branch and has been rebased on top of the latest master. If you're asked to make more changes, make sure you rebase onto master then too!
* [ ] your pull request is descriptively named and links to an issue number, i.e. `Fixes #123`
* [ ] **Make sure you have run the tests!!**
  * [ ] Integration test & Server unit tests - run `npm run allTest` from the server folder
  * [ ] Client unit test - run `npm run test` from client folder
  * [ ] End to End test - run `npm run test` from the root folder

Thank you!

### CODE REVIEW

**Comment by komalahuwalia** [comment]
This fixes #894 as well as #889

**Comment by komalahuwalia** [comment]
Issue no 1 and 2 have been fixed by #889 
Issue 3 is only partially fixed since we were not able to find out why the auth2 errors out on second init onwards.
For now we have only added a if check. In case the auth2 object does not exist, we show an error isntead of failing to log in.
This handling has been done in GoogleLoginButton.jsx and GoogleSignupButton.jsx in the handleClick methods

File: client/src/app/action/user.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ export function fetchCurrentUserForAppStartUp() {
     dispatch({
       type: ActionTypes.SET_USER_LOADING,
     });
-    axios.get('/current_user')
+    axios.get(`/current_user?timestamp=${new Date().getTime()}`)
```
Comment (MathuraMG) [inline client/src/app/action/user.js:96]: @komalahuwalia  apologies, I may have forgotten this, what does this change do?

**Review by MathuraMG — COMMENTED** [review]
@komalahuwalia , if you have given it a run though, looks good. Just had 1 doubt

**Comment by komalahuwalia** [comment]
axios.get(`/current_user?timestamp=${new Date().getTime()}`) - This change has been done so that a cached response of current user is not picked and we always make an API call to the backend

**Pull request merged**